components/style/invalidation/element/invalidator.rs

```diff
@@ -997,25 +1031,14 @@ impl<'a, 'b: 'a, E> InvalidationCollector<'a, 'b, E>
         //
         // NOTE: This thing is actually untested because testing it is flaky,
         // see the tests that were added and then backed out in bug 1328509.
-        if is_visited_dependent == VisitedDependent::Yes && now_context.relevant_link_found {
-            then_context.visited_handling = VisitedHandlingMode::RelevantLinkVisited;
-            let matched_then =
-                matches_selector(&dependency.selector,
-                                 dependency.selector_offset,
-                                 None,
-                                 &self.wrapper,
-                                 &mut then_context,
-                                 &mut |_, _| {});
-
-            now_context.visited_handling = VisitedHandlingMode::RelevantLinkVisited;
-            let matches_now =
-                matches_selector(&dependency.selector,
-                                 dependency.selector_offset,
-                                 None,
-                                 &self.element,
-                                 &mut now_context,
-                                 &mut |_, _| {});
-            if matched_then != matches_now {
+        if is_visited_dependent == VisitedDependent::Yes && relevant_link_found {
+            let should_account_for_dependency = self.check_dependency(
+                VisitedHandlingMode::RelevantLinkVisited,
+                dependency,
+                &mut relevant_link_found,
```

Comment:
Oh wait I see, it isn't used here but it is used in the other caller. How about passing a dummy value here instead to make it clearer?